### PR TITLE
s3 backend

### PR DESCRIPTION
**Pull request by xn**
- added fake aws keys for s3
- if ex_aws env vars are set, will add s3 backend in dev
- renamed `send_to_cdn` -> `send_to_cdns`
- added `key` map to Asset schema
- Validations and Backends can now either be configured with a module or a {module, config} tuple

### CODE REVIEW

**Review by leandrocp — APPROVED** [review]

File: config/test.exs [hunk 1]
```diff
@@ -12,3 +12,14 @@ config :beacon, Beacon.Repo,
   ownership_timeout: 1_000_000_000
 
 config :beacon, ecto_repos: [Beacon.Repo]
+
+config :ex_aws,
+  access_key_id: "AKIAIOSFODNN7EXAMPLE",
+  secret_access_key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+  region: "us-east-1"
```
Comment (TheFirstAvenger) [inline config/test.exs:19]: Can we document that these are fake values? I read the code before the PR description and had a moment 😅. I could see future readers of the code doing the same and not having the PR description as context.

Comment (leandrocp) [reply config/test.exs:19]: > Can we document that these are fake values? I read the code before the PR description and had a moment 😅. I could see future readers of the code doing the same and not having the PR description as context.

Good point 😂

Comment (xn) [reply config/test.exs:19]: Absolutely!

Comment (xn) [reply config/test.exs:19]: done!

**Pull request merged**